lib/ansible/modules/identity/ipa/ipa_subca.py

```diff
@@ -0,0 +1,246 @@
+  ipa_port:
+    description: Port of IPA server
+    required: false
+    default: 443
+  ipa_host:
+    description: IP or hostname of IPA server
+    required: false
+    default: ipa.example.com
+  ipa_user:
+    description: Administrative account used on IPA server
+    required: false
+    default: admin
+  ipa_pass:
+    description: Password of administrative user
+    required: true
+  ipa_prot:
+    description: Protocol used by IPA server
+    required: false
+    default: https
+    choices: ["http", "https"]
+  validate_certs:
+    description:
+    - This only applies if C(ipa_prot) is I(https).
+    - If set to C(no), the SSL certificates will not be validated.
+    - This should only set to C(no) used on personally controlled sites using self-signed certificates.
+    required: false
+    default: true
+'''
+
+EXAMPLES = '''
+# Ensure IPA Sub CA is present
+- ipa_subca:
+    ipa_host: spider.example.com
+    ipa_pass: Passw0rd!
+    state: present
+    subca_name: AnsibleSubCA1
+    subca_subject: 'CN=AnsibleSubCA1,O=example.com'
+    subca_desc: Ansible Sub CA
+
+# Ensure that IPA Sub CA is removed
+- ipa_subca:
+    ipa_host: spider.example.com
+    ipa_pass: Passw0rd!
+    state: absent
+    subca_name: AnsibleSubCA1
+
+# Ensure that IPA Sub CA is disabled
+- ipa_subca:
+    ipa_host: spider.example.com
+    ipa_pass: Passw0rd!
+    state: disable
+    subca_name: AnsibleSubCA1
+'''
+
+RETURN = '''
+subca:
+  description: IPA Sub CA record as returned by IPA API.
+  returned: always
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ipa import IPAClient
+from ansible.module_utils._text import to_native
+
+
+class SubCAIPAClient(IPAClient):
+    def __init__(self, module, host, port, protocol):
+        super(SubCAIPAClient, self).__init__(module, host, port, protocol)
+
+    def subca_find(self, subca_name):
+        return self._post_json(method='ca_find', name=subca_name, item=None)
+
+    def subca_add(self, subca_name=None, subject_dn=None, details=None):
+        item = dict(ipacasubjectdn=subject_dn)
+        subca_desc = details.get('description', None)
+        if subca_desc is not None:
+            item.update(description=subca_desc)
+        return self._post_json(method='ca_add', name=subca_name, item=item)
+
+    def subca_mod(self, subca_name=None, diff=None, details=None):
+        item = get_subca_dict(details)
+        for change in diff:
+            update_detail = dict()
+            if item[change] is not None:
+                update_detail.update(setattr="{0}={1}".format(change, item[change]))
+                self._post_json(method='ca_mod', name=subca_name, item=update_detail)
+
+    def subca_del(self, subca_name=None):
+        return self._post_json(method='ca_del', name=subca_name)
+
+    def subca_disable(self, subca_name=None):
+        return self._post_json(method='ca_disable', name=subca_name)
```

Comment:
It doesn't look like FreeIPA has these enable/disable API functions:
<img width="258" alt="screen shot 2017-08-30 at 11 57 07 am" src="https://user-images.githubusercontent.com/6355826/29884883-708d8f68-8d7a-11e7-98e3-b085a4523937.png">

And the below sample run seems to confirm:
```
fatal: [localhost -> 192.168.200.12]: FAILED! => {"changed": false, "failed": true, "msg": "response ca_enable: unknown command 'ca_enable'"}
```

Reply:
Also, for posterity:
```
$ ipa --version
VERSION: 4.4.0, API_VERSION: 2.213
```

Reply:
I think it is added after 4.4, I will confirm and add a block to check version of IPA.

Reply:
Ah ok. Yeah, after I posted, I was like "Oh wait this may just be a version thing"

Reply:
I have added check for IPA version check. Could you please review it?